cmd/acioperator/main.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2016 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	log "github.com/Sirupsen/logrus"
+	operators "github.com/noironetworks/aci-containers/pkg/acicontainersoperator/apis/aci.ctrl/v1alpha1"
+	operatorclientset "github.com/noironetworks/aci-containers/pkg/acicontainersoperator/clientset/versioned"
+	"github.com/noironetworks/aci-containers/pkg/acioperator"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+// Create K8s Client
+func getOperatorClient() operatorclientset.Interface {
+
+	restconfig, err := restclient.InClusterConfig()
+	if err != nil {
+		return nil
+	}
+
+	kubeClient, err := operatorclientset.NewForConfig(restconfig)
+	if err != nil {
+		log.Fatalf("Failed to intialize kube client %v", err)
+	}
+
+	log.Info("Successfully constructed k8s client")
+	return kubeClient
+}
+
+
+func main() {
+	// get the Kubernetes client for connectivity
+	log.Debug("Initializing kubernetes client")
+	client := getOperatorClient()
+
+	informer := cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(options meta_v1.ListOptions) (runtime.Object, error) {
+				return client.AciV1alpha1().AciContainersOperators("aci-containers-system").List(options)
+			},
+			WatchFunc: func(options meta_v1.ListOptions) (watch.Interface, error) {
+				return client.AciV1alpha1().AciContainersOperators("aci-containers-system").Watch(options)
+			},
+		},
+		&operators.AciContainersOperator{},
+		0,
+		cache.Indexers{},
+	)
+
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
this cmd file right you can register the event handlers in  operator file right, the way we are following for controller and hostagent. if you want to handle it different commit that is fine